File: algorithm/java/src/main/java/com/liyongquan/weeklycontest/bwc125/MinOperations.java
package com.liyongquan.weeklycontest.bwc125;

import java.util.Arrays;
import java.util.Collections;
import java.util.Comparator;
import java.util.PriorityQueue;

public class MinOperations {
    public int minOperations(int[] nums, int k) {
        Arrays.sort(nums);
        PriorityQueue<Integer> pq = new PriorityQueue<>((o1, o2) -> o1 - o2);
        for (int num : nums) {
            pq.add(num);
        }
        int res = 0;
        while (pq.size() > 1 && pq.peek() < k) {
            Integer num1 = pq.poll();
            Integer num2 = pq.poll();
            pq.offer(num1 * 2 + num2);
            res++;
        }
        return res;
    }
}

File: algorithm/golang/weekcontest/bwc145/minOperations.go
package bwc145

import "sort"

func minOperations(nums []int, k int) int {
	sort.Ints(nums)
	if k > nums[0] {
		return -1
	}
	mp := make(map[int]int)
	for _, num := range nums {
		mp[num]++
	}
	if k == nums[0] {
		return len(mp) - 1
	} else {
		return len(mp)
	}
}
